### PR TITLE
add KUBECTL_SHA256 passthru from JJB

### PR DESCRIPTION
**Pull request by tuminoid**
We need to pass `KUBECTL_SHA256` matching `KUBERNETES_VERSION` or `KUBERNETES_VERSION_UPGRADE_TO` version from JJB, depending on the test.

Depends on https://github.com/metal3-io/cluster-api-provider-metal3/pull/1178 and https://github.com/metal3-io/metal3-dev-env/pull/1289 to been merged first.

### CODE REVIEW

**Comment by tuminoid** [comment]
This should not fail, since capm3 PR is not merged yet, but let's see.

/test-ubuntu-integration-main

**Comment by tuminoid** [comment]
/test-centos-e2e-integration-main

This e2e should fail with an error message specified in this PR as fake digest.

**Comment by tuminoid** [comment]
> /test-centos-e2e-integration-main
> 
> This e2e should fail with an error message specified in this PR as fake digest.

Actually it should not, since the versions match, the digest isn't actually going to be checked.

/test-e2e-upgrade-main-from-release-0-5
/test-e2e-upgrade-main-from-release-1-4

**Comment by tuminoid** [comment]
```console
fatal: https://storage.googleapis.com/kubernetes-release/release/v1.25.5/bin/linux/amd64/kubectl binary checksum '6a660cd44db3d4bfe1563f6689cbe2ffb28ee4baf3532e04fff2d7b909081c29' differs from expected checksum 'this-was-defined-in-integration-test-script'
```

All checks out, should be fine to pass thru from JJB.

**Comment by tuminoid** [comment]
/test-ubuntu-integration-main 

/hold
I'll add the SHA digests into JJB first, so we can test the passthru.

**Comment by tuminoid** [comment]
One more JJB change needed, waiting that.

**Comment by tuminoid** [comment]
/unhold

Upgrade tests have passed the part wherre it checks for kubectl sha, so it is going to work.

**Review by lentzi90 — COMMENTED** [review]
/approve
Thanks for the thorough tests and explanations, really helpful when reviewing!

**Comment by tuminoid** [comment]
/test-ubuntu-integration-main  

Fixed WIP in git commit title, and removed one placeholder text.

File: jenkins/scripts/files/run_integration_tests.sh [hunk 1]
```diff
@@ -28,14 +28,17 @@ if [[ "${REPO_NAME}" == "metal3-dev-tools" ]]; then
...
 fi
 
 if [[ "${CAPM3_VERSION}" == "v1alpha5" ]]; then
     export KUBERNETES_VERSION="v1.23.8"
+    export KUBECTL_SHA256="${KUBECTL_SHA256:-4685bfcf732260f72fce58379e812e091557ef1dfc1bc8084226c7891dd6028f}"
 fi
 
 if [[ "${GINKGO_FOCUS}" == "k8s-upgrade" ]]; then
     export KUBERNETES_VERSION="${KUBERNETES_VERSION_UPGRADE_TO}"
+    export KUBECTL_SHA256="${KUBECTL_SHA256:-}"
```
Comment (kashifest) [inline jenkins/scripts/files/run_integration_tests.sh:41]:
```suggestion
    export KUBECTL_SHA256="${KUBECTL_SHA256}"
```

**Review by kashifest — CHANGES_REQUESTED** [review]

**Comment by tuminoid** [comment]
/test-e2e-1-25-1-26-upgrade-main
/test-e2e-1-26-1-27-upgrade-main
/test-e2e-1-27-1-28-upgrade-main

Let's make sure the last change didn't affect these. They're good if they pass the kubectl sha verification, even if they fail later.

**Review by kashifest — APPROVED** [review]
/lgtm

**Comment by tuminoid** [comment]
/test-ubuntu-integration-main
Db locked flake.

**Pull request merged**